### PR TITLE
Add dynamic parameter support to deployment stack New/Set cmdlets.

### PR DESCRIPTION
**Pull request by dantedallag**
## Description

Added dynamic parameters to load template parameters from provided user template. This feature is already available in deployments cmdlets and a similar pattern was used. Also involved some refactoring.

## Mandatory Checklist

- Please choose the target release of Azure PowerShell
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] N/A

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Resources/Resources/ChangeLog.md [hunk 1]
```diff
@@ -23,6 +23,7 @@
 * Fixed bug with custom types and deployments whatif. [#13245]
 * Fixed bug with nullable array parameters & outputs.
 * Fixed bug with TemplateParameterUri not downloading parameters correctly.
+* Add dynmaic parameters to stack New/Set cmdlets.
```
Comment (vidai-msft) [inline src/Resources/Resources/ChangeLog.md:26]: Please use past tense. 
```suggestion
* Added dynamic parameters to stack New/Set cmdlets.
```

**Review by vidai-msft — CHANGES_REQUESTED** [review]

**Comment by vidai-msft** [comment]
@dantedallag Please rebase your branch with the latest main since ChangeLog.md has been updated. Please also make sure the new change log item appears at the top of the Upcoming Release section. Thanks!

File: src/Resources/ResourceManager/Implementation/CmdletBase/DeploymentStacksCreateCmdletBase.cs [hunk 2]
```diff
@@ -138,56 +252,129 @@ protected DeploymentStacksCreateCmdletBase()
...
+                string.IsNullOrEmpty(TemplateUri) &&
                 string.IsNullOrEmpty(TemplateSpecId))
             {
                 // When .bicepparam support was first introduced, we were missing the validation to block overriding the 'using' path in these cmdlets.
                 // We need to be careful to retain this behavior to avoid breaking existing users, until it is re-introduced intentionally with https://github.com/Azure/bicep/issues/10333.
 
                 if (!string.IsNullOrEmpty(output.templateJson))
                 {
-                    TemplateJson = output.templateJson;
+                    TemplateObject = JsonConvert.DeserializeObject<Hashtable>(output.templateJson);
```
Comment (vidai-msft) [inline src/Resources/ResourceManager/Implementation/CmdletBase/DeploymentStacksCreateCmdletBase.cs:377]: @anthony-c-martin Yesterday I merged the PR [#24282] and you replaced JsonConvert.DeserializeObject<Hashtable>(output.templateJson) to output.templateJson.FromJson<Hashtable>(). Could you please confirm if it should apply to this line of code?

File: src/Resources/ResourceManager/Implementation/CmdletBase/DeploymentStacksCreateCmdletBase.cs [hunk 3]
```diff
@@ -196,11 +383,21 @@ protected Hashtable ResolveParameters()
...
+        }
 
-            return output.parameters;
+        /// <summary>
+        /// Constructs a TemplateObject from the bicep file located at the TemplateFile address. 
+        /// </summary>
+        protected void BuildAndUseBicepTemplate()
+        {
+            var templateJson = BicepUtility.Create().BuildBicepFile(this.ResolvePath(TemplateFile), this.WriteVerbose, this.WriteWarning);
+            TemplateObject = JsonConvert.DeserializeObject<Hashtable>(templateJson);
```
Comment (vidai-msft) [inline src/Resources/ResourceManager/Implementation/CmdletBase/DeploymentStacksCreateCmdletBase.cs:399]: cc @anthony-c-martin

File: src/Resources/ResourceManager/Implementation/CmdletBase/DeploymentStacksCreateCmdletBase.cs (hunk 2 above)
Comment (anthony-c-martin) [reply src/Resources/ResourceManager/Implementation/CmdletBase/DeploymentStacksCreateCmdletBase.cs:377]: Yes they'll need to be integrated here - I'm guessing this was missed in a rebase. @dantedallag please could you incorporate them?

**Review by vidai-msft — APPROVED** [review]
LGTM

**Pull request merged**